lib/WeBWorK3/Hooks.pm

```diff
@@ -43,10 +41,17 @@ sub has_permission ($user, $perm) {
 # Check permission for /api routes
 our $check_permission = sub ($next, $c, $action, $) {
 	return $next->() if ($c->ignore_permissions || $c->req->url->to_string =~ /\/api\/login/x);
+	my $user = $c->current_user;
+	if ($c->stash('course_id')) {
+		my $user_obj    = $c->schema->resultset('User')->getGlobalUser({ user_id => $c->current_user->{user_id} }, 1);
```

Comment:
This is the problem.  This should be:
```suggestion
		my $user_obj = $c->schema->resultset('User')
			->getGlobalUser(info => { user_id => $c->current_user->{user_id} }, as_result_set => 1);
```
I don't know how this could possibly be working as it is for you @drdrew42.